ports/re2/README.md

```diff
@@ -0,0 +1,202 @@
+git clone https://github.com/qnx-ports/googletest.git
+git clone https://github.com/qnx-ports/benchmark.git
+git clone https://github.com/qnx-ports/muslflt.git
+git clone https://github.com/qnx-ports/abseil-cpp.git
+git clone https://github.com/qnx-ports/re2.git
+
+#Via SSH
+git clone git@github.com:qnx-ports/build-files.git 
+git clone git@github.com:qnx-ports/googletest.git
+git clone git@github.com:qnx-ports/benchmark.git
+git clone git@github.com:qnx-ports/muslflt.git
+git clone git@github.com:qnx-ports/abseil-cpp.git
+git clone git@github.com:qnx-ports/re2.git 
+```
+
+3. Build the Docker image and create a container
+```bash
+cd build-files/docker
+./docker-build-qnx-image.sh
+./docker-create-container.sh
+```
+
+4. Source your SDP (Installed from QNX Software Center)
+```bash
+#QNX 8.0 will be in the directory ~/qnx800/
+#QNX 7.1 will be in the directory ~/qnx710/
+source ~/qnx800/qnxsdp-env.sh
+```
+
+5. Build dependencies in order in your workspace from Step 1. Skip those you already have installed.
+```bash
+#1. googletest
+BUILD_TESTING="OFF" QNX_PROJECT_ROOT="$(pwd)/googletest" make -C build-files/ports/googletest install -j$(nproc)
+#2. benchmark
+QNX_PROJECT_ROOT="$(pwd)/benchmark" make -C build-files/ports/benchmark install -j$(nproc)
+#3. muslflt
+QNX_PROJECT_ROOT="$(pwd)/muslflt" make -C build-files/ports/muslflt/ install -j$(nproc)
+#4. abseil-cpp
+BUILD_TESTING="OFF" QNX_PROJECT_ROOT="$(pwd)/abseil-cpp" make -C build-files/ports/abseil-cpp install -j$(nproc)
+```
+
+
+6. Build the project in your workspace from Step 1
+```bash
+# Do not set QNX_BUILD_TESTS to anything if you do not want tests built.
+# With Tests
+OSLIST=nto QNX_BUILD_TESTS="yes" QNX_PROJECT_ROOT="$(pwd)/re2" make -C build-files/ports/re2 install -j$(nproc)
+# Without Tests
+OSLIST=nto QNX_PROJECT_ROOT="$(pwd)/re2" make -C build-files/ports/re2 install -j$(nproc)
+```
+
+**NOTE**: Before rebuilding, you may need to delete the `/build` subdirectories and their contents in `build-files/ports/memory/nto-aarch64-le` and `build-files/ports/re2/nto-x86_64-o`. This MUST be done when changing from SDP 7.1 to 8 or vice versa, as it will link against the wrong shared objects and not show an error until testing.
+```bash
+#From your workspace:
+make -C build-files/ports/re2 clean 
+```
+
+# Running Tests on a Target
+Some distributions of QNX have critical directories stored in a read-only partition (`/`, `/system`, `/etc`, etc). Included in these are the default `bin` and `lib` directories. If this is the case, follow the "Installing in home directory" instructions
+
+
+
+### Installing in home directory
+1. Installation
+```bash
+#Set your target's IP here
+TARGET_IP_ADDRESS=<target-ip-address-or-hostname>
+
+#Select the home directory to install to (this will install to /data/home/qnxuser)
+TARGET_USER_FOR_INSTALL="qnxuser"
+
+#Create new directories on the target
+ssh qnxuser@$TARGET_IP_ADDRESS "mkdir -p /data/home/$TARGET_USER_FOR_INSTALL/memory/lib"
```

Comment:
Creating the wrong directory

Reply:
Good catch, I noticed a few other things wrong in the testing section - resolved with 4abeb962f4badac62791eb919c545aabb19d9665 !